From a24e3486a85ea5f417c5c72acbf25f7e21f7e696 Mon Sep 17 00:00:00 2001
From: Dhaval Khandla <dhavaljk@ti.com>
Date: Fri, 31 Jan 2020 18:05:23 +0530
Subject: [PATCH 3/6] arm64: dts: ti: Use a bigger region for PCIe outbound
 addresses

Some applications need to expose more than 128MB over PCIe, which
is limited by the default outbound address spaces being 128MB. This
commit modifies the default region for allocating PCIe outbound
addresses to a 4GB region. Out of 4GB, it has been tested that upto 2GB
allocation works in host mode, while whole 4GB works in endpoint mode.
So this commit makes appropriate changes for host and endpoint mode.

Signed-off-by: Dhaval Khandla <dhavaljk@ti.com>
---
 arch/arm64/boot/dts/ti/k3-j721e-main.dtsi | 8 ++++----
 arch/arm64/boot/dts/ti/k3-j721e.dtsi      | 2 ++
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi b/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi
index 4e04d5b14521..8260d323c164 100644
--- a/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi
@@ -1004,13 +1004,13 @@
 			iommu-map = <0x0 &smmu0 0x0 0x1000>;
 			dma-coherent;
 			ranges = <0x01000000 0x0 0x10001000  0x00 0x10001000  0x0 0x0010000>,
-				 <0x02000000 0x0 0x10011000  0x00 0x10011000  0x0 0x7fef000>;
+				 <0x02000000 0x0 0x00000000  0x40 0x00000000  0x0 0x80000000>;
 		};
 
 		pcie0_ep: pcie-ep@d000000 {
 			compatible = "ti,j721e-cdns-pcie-ep";
 			reg = <0x00 0x0d000000 0x00 0x00800000>,
-			      <0x00 0x10000000 0x00 0x08000000>;
+			      <0x40 0x00000000 0x01 0x00000000>;
 			reg-names = "reg", "addr_space";
 			cdns,max-outbound-regions = <16>;
 			max-functions = /bits/ 8 <6>;
@@ -1052,13 +1052,13 @@
 			iommu-map = <0x0 &smmu0 0x04000 0x1000>;
 			dma-coherent;
 			ranges = <0x01000000 0x0 0x18001000  0x00 0x18001000  0x0 0x0010000>,
-				 <0x02000000 0x0 0x18011000  0x00 0x18011000  0x0 0x7fef000>;
+				 <0x02000000 0x0 0x00000000  0x41 0x00000000  0x0 0x80000000>;
 		};
 
 		pcie1_ep: pcie-ep@d800000 {
 			compatible = "ti,j721e-cdns-pcie-ep";
 			reg = <0x00 0x0d800000 0x00 0x00800000>,
-			      <0x00 0x18000000 0x00 0x08000000>;
+			      <0x41 0x00000000 0x01 0x00000000>;
 			reg-names = "reg", "addr_space";
 			cdns,max-outbound-regions = <16>;
 			max-functions = /bits/ 8 <6>;
diff --git a/arch/arm64/boot/dts/ti/k3-j721e.dtsi b/arch/arm64/boot/dts/ti/k3-j721e.dtsi
index 279d17689139..b9d48e08d4ff 100644
--- a/arch/arm64/boot/dts/ti/k3-j721e.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-j721e.dtsi
@@ -146,6 +146,8 @@
 			 <0x00 0x0d000000 0x00 0x0d000000 0x00 0x01800000>, /* PCIe Core*/
 			 <0x00 0x10000000 0x00 0x10000000 0x00 0x10000000>, /* PCIe DAT */
 			 <0x00 0x64800000 0x00 0x64800000 0x00 0x00800000>, /* C71 */
+			 <0x40 0x00000000 0x40 0x00000000 0x01 0x00000000>, /* PCIe2 DAT */
+			 <0x41 0x00000000 0x41 0x00000000 0x01 0x00000000>, /* PCIe2 DAT */
 			 <0x44 0x00000000 0x44 0x00000000 0x00 0x08000000>, /* PCIe2 DAT */
 			 <0x4d 0x80800000 0x4d 0x80800000 0x00 0x00800000>, /* C66_0 */
 			 <0x4d 0x81800000 0x4d 0x81800000 0x00 0x00800000>, /* C66_1 */
-- 
2.17.1

